components/script/dom/document.rs

```diff
@@ -1599,17 +1599,27 @@ impl Document {
         // Step 5 can be found in asap_script_loaded and
         // asap_in_order_script_loaded.
 
-        if self.loader.borrow().is_blocked() {
+        let loader = self.loader.borrow();
+        if loader.is_blocked() || loader.events_inhibited() || !self.has_browsing_context {
```

Comment:
This is necessary for cases that involve setting innerHTML. ServoHTMLParser::parse_html_fragment creates a new Document, which has an associated window but no browsing context. When the parser finishes, it calls Document::finish_load which then adds the associated window's pipeline to the list of documents that may be ready for load events. The document that is retrieved at the end of the event loop is a different document than the one that had finish_load called on it, and this means we can end up hitting the assertion in maybe_queue_document_completion. I think as written these changes are incorrect, because the document that finishes being parsed will never be marked complete.